crates/autopilot/src/decoded_settlement.rs

```diff
@@ -360,7 +348,7 @@ impl DecodedSettlement {
                 let order = orders.swap_remove(i);
 
                 // Update fee only for orders with solver computed fees (limit orders)
-                if !order.solver_determines_fee {
+                if matches!(order.executed_fee, ExecutedFee::Solver { .. }) {
```

Comment:
In conjunction with the comment above this reads counter intuitive (since we are not updating the fee for `ExecutedFee::Solver { .. }` despite saying so). I know that this still captures limit orders, but let's maybe change this to make sense when someone with less context reads these two lines.

Ie.

```rs
// Update fee only for orders where fee is taken from surplus
if !matches!(order.executed_fee, ExecutedFee::Surplus) {
  return None
}
```

Maybe we can also rename `ExecutedFee::Solver` to `ExecutedFee::Order` because it's not really the solver that is specifying this fee (but instead the user order)

Reply:
Renamed `ExecutedFee::Solver` to `ExecutedFee::Order`.
The `if` line is about to be deleted in the next cleanup PR but i still fixed it with your proposal. In the next cleanup PR I will "calculate" the fee for market orders also and save all the order executions from `OnSettlementEventUpdater`.